lib/internal/async_hooks.js

```diff
@@ -212,23 +212,23 @@ function storeActiveHooks() {
   // Don't want to make the assumption that kInit to kDestroy are indexes 0 to
   // 4. So do this the long way.
   active_hooks.tmp_fields = [];
-  active_hooks.tmp_fields[kInit] = async_hook_fields[kInit];
-  active_hooks.tmp_fields[kBefore] = async_hook_fields[kBefore];
-  active_hooks.tmp_fields[kAfter] = async_hook_fields[kAfter];
-  active_hooks.tmp_fields[kDestroy] = async_hook_fields[kDestroy];
-  active_hooks.tmp_fields[kPromiseResolve] = async_hook_fields[kPromiseResolve];
+  setHooks(active_hooks.tmp_fields, async_hook_fields);
+}
+
+function setHooks(from, to) {
```

Comment:
I'm not sure if I like these kinds of abstractions, but for sure it seems that `from` and `to` should be switched.

Reply:
Maybe call it `copyHooks`? Maybe it’s just my C background that sees parallels to `memcpy()` & co here, but it seems closer to what’s happening than `setHooks`?

Reply:
I like `copyHooks`, I'll update the PR. Thanks

Reply:
@AndreasMadsen  I'll also changed the parameters, so it will be `copyHooks(destination, source)`.